File: Data Structures & Algorithms/insert-new-interval/submission-1.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        intervals.append(newInterval)
        intervals.sort()

        newIntervals = []

        for i in intervals:
            if len(newIntervals) == 0:
                newIntervals.append(i)
            else:
                cur = newIntervals.pop()
                if cur[1] >= i[0]:
                    newIntervals.append([cur[0], max(cur[1], i[1])])
                else:
                    newIntervals.append(cur)
                    newIntervals.append(i)

        return newIntervals

File: Data Structures & Algorithms/meeting-schedule-ii/submission-1.js
/**
 * Definition of Interval:
 * class Interval {
 *   constructor(start, end) {
 *     this.start = start;
 *     this.end = end;
 *   }
 * }
 */

class Solution {
    /**
     * @param {Interval[]} intervals
     * @returns {number}
     */
    minMeetingRooms(intervals) {
        const n = intervals.length;
        if (n == 0) return 0;

        let res = 0;

        const starts = intervals.map(iv => iv.start).sort((a,b)=>a-b);
        const ends = intervals.map(iv => iv.end).sort((a,b)=>a-b);

        let s = 0;
        let e = 0;

        while (s < n) {
            if (starts[s] < ends[e]) {
                s++;
            } else {
                e++;
            }
            res = Math.max(res, s - e);
        }

        return res;
    }
}
